mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,754 @@
+                    };
+
+                    let details = match msg.type_url.as_str() {
+                        SEND_TYPE_URL => match parse_send_tx_details(args).await {
+                            Ok(details) => details,
+                            Err(e) => {
+                                return Err(Stopped {
+                                    phantom: Default::default(),
+                                    stop_reason: e,
+                                });
+                            },
+                        },
+                        CLAIM_HTLC_TYPE_URL => match parse_claim_htlc_tx_details(args).await {
+                            Ok(details) => details,
+                            Err(e) => {
+                                return Err(Stopped {
+                                    phantom: Default::default(),
+                                    stop_reason: e,
+                                });
+                            },
+                        },
+                        CREATE_HTLC_TYPE_URL => match parse_create_htlc_tx_details(args).await {
+                            Ok(details) => details,
+                            Err(e) => {
+                                return Err(Stopped {
+                                    phantom: Default::default(),
+                                    stop_reason: e,
+                                });
+                            },
+                        },
+                        _ => continue,
+                    };
+
+                    let details = match details {
+                        Some(details) => details,
+                        None => continue,
+                    };
+
+                    upsert_tx_details(&mut tx_details, details);
+                }
+
+                for (ticker, txs) in tx_details.into_iter() {
+                    let id = WalletId::new(ticker.replace('-', "_"));
+
+                    try_or_return_stopped_as_err!(
+                        storage
+                            .add_transactions_to_history(&id, txs)
+                            .await
+                            .map_err(|e| format!("{:?}", e)),
+                        StopReason::StorageError,
+                        "add_transactions_to_history failed"
+                    );
+                }
+
+                iterate_more = (TX_PAGE_SIZE as u32 * page) < response.total_count;
+                page += 1;
+            }
+
+            Ok(())
+        }
+
+        let q = format!("transfer.sender = '{}'", self.address.clone());
+        if let Err(stopped) = fetch_and_insert_txs(&ctx.coin, &ctx.storage, q, &self.active_assets).await {
+            return Self::change_state(stopped);
+        };
+
+        let q = format!("transfer.recipient = '{}'", self.address.clone());
+        if let Err(stopped) = fetch_and_insert_txs(&ctx.coin, &ctx.storage, q, &self.active_assets).await {
+            return Self::change_state(stopped);
+        };
```

Comment:
If there are transactions from and to the same address, we'll do the work twice. Is there a way to handle such case?

Reply:
Not really. You can't filter out some addresses from tx search. For example if we get 10 transactions from tx_search, and 4th transaction is self made, you can't filter it out on the query. You can skip it while processing the data and that's already implemented.